internal/proxy/configuration.go

```diff
@@ -0,0 +1,458 @@
+package proxy
+
+import (
+	"encoding/base64"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/micro/go-micro/config"
+	"github.com/micro/go-micro/config/source/env"
+	"github.com/mitchellh/mapstructure"
+	"golang.org/x/xerrors"
+)
+
+// DefaultProxyCongig specifies all the defaults used to configure sso-proxy
+// All configuration can be set using environment variables. Below is a list of
+// configuration variables via their environment configuraiton
+//
+// SESSION_COOKIE_NAME
+// SESSION_COOKIE_SECRET
+// SESSION_COOKIE_EXPIRE
+// SESSION_COOKIE_DOMAIN
+// SESSION_COOKIE_HTTPONLY
+// SESSION_TTL_LIFETIME
+// SESSION_TTL_VALID
+// SESSION_TTL_GRACEPERIOD
+//
+// REQUESTSIGNER_KEY
+//
+// CLIENT_ID
+// CLIENT_SECRET
+//
+// SERVER_PORT
+// SERVER_TIMEOUT_SHUTDOWN
+// SERVER_TIMEOUT_READ
+// SERVER_TIMEOUT_WRITE
+//
+// METRICS_STATSD_HOST
+// METRICS_STATSD_PORT
+//
+// LOGGING_ENABLE
+// LOGGING_LEVEL
+//
+// UPSTREAM_DEFAULT_EMAIL_DOMAINS
+// UPSTREAM_DEFAULT_EMAIL_ADDRESSES
+// UPSTREAM_DEFAULT_GROUPS
+// UPSTREAM_DEFAULT_TIMEOUT
+// UPSTREAM_DEFAULT_TCP_RESET_DEADLINE
+// UPSTREAM_DEFAULT_PROVIDER
+// UPSTREAM_CONFIGS_FILE
+// UPSTREAM_SCHEME
+// UPSTREAM_CLUSTER
+//
+// PROVIDER_TYPE
+// PROVIDER_URL_EXTERNAL
+// PROVIDER_URL_INTERNAL
+// PROVIDER_SCOPE
+
+func DefaultProxyConfig() Configuration {
+	return Configuration{
+		ServerConfig: ServerConfig{
```

Comment:
Considering the name of this package is already called `configuration` and it's pulled in/referenced elsewhere as `config` I don't believe we need individual fields to also include 'config' in their name.

It reads fine in most cases, e.g. `config.LoggingConfig` actually kinda _sounds_ better than `config.Logging` but strictly speaking it feels like it should be redundant.

But on the flip-side that 'sounding better' argument gets weaker when we get to examples like `config.ServerConfig.TimeoutConfig` vs `config.Server.Timeout`.

Reply:
Interesting. I think in most cases you're right, though because they are their own independent, separate structs I'm not sure how well it would read in the receiving functions, e.g:
`func SetCookieStore(cc CookieConfig) func(*OAuthProxy) error {` 
vs
`func SetCookieStore(cc Cookie) func(*OAuthProxy) error {` 
or in some tests. I need to have a look closer look through.

I think this is a good point and definitely worth considering, though I think I'd personally opt to look at this in a separate followup pull request - and look at this and `sso_auth`'s equivalent configuration package _together_ in this aspect.

Let me know if you have any objections there!